packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,500 @@
+/// another [Navigator] setup is needed, like for example to enamble named routes
+/// or the pages API, then it is recommended to directly push a [CupertinoSheetRoute]
+/// to the stack with whatever configuration needed. See the implementation below
+/// for the boilerplate provided by `showCupertinoSheet` to get started.
+///
+/// The whole sheet can be popped at once by either dragging down on the shet,
+/// or calling [CupertinoSheetRoute.popSheet].
+///
+/// iOS sheet widgets are generally designed to be tightly coupled to the context
+/// of the widget that opened the sheet. As such, it is not recommended to directly
+/// link to a route outside of the sheet, without first popping the sheet. If that
+/// is needed however, it can be done by pushing to the root Navigator.
+///
+/// If `useNestedNavigation` is left as `false`, then a [CupertinoSheetRoute]
+/// will be shown with no [Navigator] widget. Multiple calls to `showCupertinoSheet`
+/// can still be made to show multiple stacked sheets, if desired.
+///
+/// In both cases, new [CupertinoSheetRoute]s will be pushed to the root
+/// [Navigator]. This is to ensure the previous routes animate correctly.
+///
+/// Returns a [Future] that resolves to the value (if any) that was passed to
+/// [Navigator.pop] when the sheet was closed.
+///
+/// {@tool dartpad}
+/// This example shows how to navigate to use [showCupertinoSheet] to display a
+/// Cupertino sheet widget with nested navigation.
+///
+/// ** See code in examples/api/lib/cupertino/sheet/cupertino_sheet.2.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [CupertinoSheetRoute] the basic route version of the sheet view.
+///  * [showCupertinoDialog] which displays an iOS-styled dialog.
+Future<T?> showCupertinoSheet<T>({
+  required BuildContext context,
+  required WidgetBuilder pageBuilder,
+  bool useNestedNavigation = false,
+}) {
+  final WidgetBuilder builder;
+  if (!useNestedNavigation) {
+    builder = pageBuilder;
+  } else {
+    builder = (BuildContext context) {
+      return NavigatorPopHandler(
+        onPopWithResult: (T? result) {
+          Navigator.of(context, rootNavigator: true).maybePop(result);
+        },
+        child: Navigator(
+          initialRoute: '/',
+          onGenerateRoute: (RouteSettings settings) {
+            return CupertinoPageRoute<void>(
+              builder: (BuildContext context) {
+                return PopScope(
+                  canPop: settings.name != '/',
+                  onPopInvokedWithResult: (bool didPop, Object? result) {
+                    if (didPop) {
+                      return;
+                    }
+                    Navigator.of(context, rootNavigator: true).pop(result);
+                  },
+                  child: pageBuilder(context),
+                );
+              }
+            );
+          },
+        ),
+      );
+    };
+  }
+
+  return Navigator.of(context, rootNavigator: true).push<T>(CupertinoSheetRoute<T>(
```

Comment:
useRootNavigator was removed because we always want to go to the root when adding a new sheet. The sheet transition does not look correct if it tries to animate within a nested navigator.

Reply:
sgtm